### ISSUE TITLE
No credentials provided issue.

### ISSUE THREAD

**Issue by ipolyzos** [CiscoCloud/mantl#1021]
Any attempt for frameworks or even mesos cli to communicate with the cluster result into a message such as:

```
I0125 10:30:57.357239 27386 sched.cpp:272] No credentials provided. Attempting to register without authentication
```
this has been seen when try to utilize spark jobs, register kafka & cassandra even when tried to manually execute a cli i.e.

```
 mesos execute --command="echo 'hello mesos'" --name='HelloMesos' --master=$(mesos resolve zk://XXX.XXX.XXX.XXX:2181/mesos)
```

Could you please advise?

**Comment by ipolyzos** [comment]
the logs from ```mesos execute```

**```> mesos execute --command="echo 'hello mesos'" --name='HelloMesos' --master=$(mesos resolve zk://XXX.XXX.XXX.XXX:2181/mesos)```**

```
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@712: Client environment:zookeeper.version=zookeeper C client 3.4.5
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@716: Client environment:host.name=XXX
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@723: Client environment:os.name=Linux
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@724: Client environment:os.arch=3.10.0-229.el7.x86_64
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@725: Client environment:os.version=#1 SMP Fri Mar 6 11:36:42 UTC 2015
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@733: Client environment:user.name=centos
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@741: Client environment:user.home=/home/centos
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@log_env@753: Client environment:user.dir=XXX
2016-01-25 10:30:53,727:27367(0x7fda4bc07700):ZOO_INFO@zookeeper_init@786: Initiating client connection, host=XXX.XXX.XXX.XXX:2181 sessionTimeout=10000 watcher=0x7fda5389b160 sessionId=0 sessionPasswd=<null> context=0x7fda28001b60 flags=0
2016-01-25 10:30:53,772:27367(0x7fda46d7b700):ZOO_INFO@check_events@1703: initiated connection to server [10.104.50.172:2181]
2016-01-25 10:30:53,776:27367(0x7fda46d7b700):ZOO_INFO@check_events@1750: session establishment complete on server [10.104.50.172:2181], sessionId=0x1521d33150b0032, negotiated timeout=10000
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0125 10:30:53.777542 27371 group.cpp:331] Group process (group(1)@10.19.148.4:58487) connected to ZooKeeper
I0125 10:30:53.777581 27371 group.cpp:805] Syncing group operations: queue size (joins, cancels, datas) = (0, 0, 0)
I0125 10:30:53.777588 27371 group.cpp:379] Authenticating with ZooKeeper using digest
2016-01-25 10:30:57,109:27367(0x7fda46d7b700):ZOO_INFO@auth_completion_func@1286: Authentication scheme digest succeeded
I0125 10:30:57.109812 27371 group.cpp:403] Trying to create path '/mesos' in ZooKeeper
I0125 10:30:57.111749 27371 detector.cpp:156] Detected a new leader: (id='5')
I0125 10:30:57.111847 27375 group.cpp:674] Trying to get '/mesos/info_0000000005' in ZooKeeper
W0125 10:30:57.112638 27375 detector.cpp:444] Leading master master@10.10.10.4:15050 is using a Protobuf binary format when registering with ZooKeeper (info): this will be deprecated as of Mesos 0.24 (see MESOS-2340)
I0125 10:30:57.113106 27375 detector.cpp:481] A new leading master (UPID=master@10.10.10.4:15050) is detected
I0125 10:30:57.354746 27379 sched.cpp:164] Version: 0.25.0
I0125 10:30:57.357053 27386 sched.cpp:262] New master detected at master@10.10.10.4:15050
I0125 10:30:57.357239 27386 sched.cpp:272] No credentials provided. Attempting to register without authentication
E0125 10:33:04.565322 27388 socket.hpp:174] Shutdown failed on fd=6: Transport endpoint is not connected [107]
```

**Comment by ryane** [comment]
By default, mantl clusters are built with mesos authentication enabled. You can see the list of valid mesos credentials that are generated in your security.yml. Many frameworks allow you to pass in the credentials in some way - usually via cli args, a config file, or env variables. For example, the cassandra-mesos framework will read the 'DEFAULT_PRINCIPAL` and `DEFAULT_SECRET` env vars. The kafka framework has `--principal` and `--secret` cli args (see https://github.com/mesos/kafka#scheduler-configuration). Unfortunately, there doesn't seem to be a standard so you have to consult the specific framework to verify if it supports auth and how to configure it. I don't see an obvious way to pass in credentials when using `mesos execute`.

**Comment by siddharthist** [comment]
@ipolyzos Are you still having this issue?

**Comment by ipolyzos** [comment]
@siddharthist i don't have the issue anymore. 

The issue risen due to that mesos framework authentication is not supported in Spark 1.4.0 (you can refer to [SPARK-6284](https://issues.apache.org/jira/browse/SPARK-6284) for more). This has been addressed in spark-1.5.0 ( please refer to [Spark 1.5.0 - Release notes](https://spark.apache.org/releases/spark-release-1-5-0.html)). 

To work around, you can either upgrade to Spark 1.5.0 or temporarily to disable authentication (i.e. set --authenticate flag to false, for more info pleae refer to [mesos role readme.rs](https://github.com/CiscoCloud/mantl/blob/master/roles/mesos/README.rst))
